index.php

```diff
@@ -50,38 +51,39 @@
     $container['notAllowedHandler'] = [$frontController, 'notAllowed'];
 
     // Routes.
+    $basePath = ViewFactory::getBasepath($container->get('request'));
     $app->get(
-        '/',
+        $basePath . '/',
```

Comment:
this does not work but I cant seem to find out why

Reply:
This works for me:
```diff
diff --git a/classes/RouterPathMiddleware.php b/classes/RouterPathMiddleware.php
new file mode 100644
index 0000000..f131bbf
--- /dev/null
+++ b/classes/RouterPathMiddleware.php
@@ -0,0 +1,42 @@
+<?php
+
+namespace Alltube;
+
+use Psr\Container\ContainerInterface;
+use Slim\Http\Request;
+use Slim\Http\Response;
+use Slim\Interfaces\RouterInterface;
+
+/**
+ * Class RouterPathMiddleware
+ * @package Alltube
+ */
+class RouterPathMiddleware
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * RouterPathMiddleware constructor.
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->router = $container->get('router');
+    }
+
+    /**
+     * @param Request $request
+     * @param Response $response
+     * @param callable $next
+     * @return mixed
+     */
+    public function __invoke(Request $request, Response $response, callable $next)
+    {
+        $this->router->setBasePath(current($request->getHeader('X-Forwarded-Path')));
+
+        return $next($request, $response);
+    }
+}
diff --git a/index.php b/index.php
index 1113e2d..a41e7dc 100644
--- a/index.php
+++ b/index.php
@@ -10,6 +10,7 @@ use Alltube\ErrorHandler;
 use Alltube\LocaleManagerFactory;
 use Alltube\LocaleMiddleware;
 use Alltube\LoggerFactory;
+use Alltube\RouterPathMiddleware;
 use Alltube\ViewFactory;
 use Slim\App;
 use Slim\Container;
@@ -33,6 +34,7 @@ try {
     $container['locale'] = LocaleManagerFactory::create();
 
     $app->add(new LocaleMiddleware($container));
+    $app->add(new RouterPathMiddleware($container));
 
     // Smarty.
     $container['view'] = ViewFactory::create($container);
@@ -54,37 +56,37 @@ try {
     // Routes.
     $basePath = current($container->get('request')->getHeader('X-Forwarded-Path'));
     $app->get(
-        $basePath . '/',
+        '/',
         [$frontController, 'index']
     )->setName('index');
 
     $app->get(
-        $basePath . '/extractors',
+        '/extractors',
         [$frontController, 'extractors']
     )->setName('extractors');
 
     $app->any(
-        $basePath . '/info',
+        '/info',
         [$frontController, 'info']
     )->setName('info');
 
     $app->any(
-        $basePath . '/watch',
+        '/watch',
         [$frontController, 'info']
     );
 
     $app->any(
-        $basePath . '/download',
+        '/download',
         [$downloadController, 'download']
     )->setName('download');
 
     $app->get(
-        $basePath . '/locale/{locale}',
+        '/locale/{locale}',
         [$frontController, 'locale']
     )->setName('locale');
 
     $app->get(
-        $basePath . '/json',
+        '/json',
         [$jsonController, 'json']
     )->setName('json');
 ```

(It has to be in a middleware or it will be set too early and then overridden by Slim.)

Reply:
applied your suggestion

thanks for the hint regarding the too early!

Reply:
Sorry, I made a mistake in the type hint.

This should fix it:
```diff
diff --git a/classes/RouterPathMiddleware.php b/classes/RouterPathMiddleware.php
index f131bbf..71e2314 100644
--- a/classes/RouterPathMiddleware.php
+++ b/classes/RouterPathMiddleware.php
@@ -5,7 +5,7 @@ namespace Alltube;
 use Psr\Container\ContainerInterface;
 use Slim\Http\Request;
 use Slim\Http\Response;
-use Slim\Interfaces\RouterInterface;
+use Slim\Router;
 
 /**
  * Class RouterPathMiddleware
@@ -14,7 +14,7 @@ use Slim\Interfaces\RouterInterface;
 class RouterPathMiddleware
 {
     /**
-     * @var RouterInterface
+     * @var Router
      */
     private $router;
```

Reply:
fixed